### PR TITLE
Formatted .py files according to PEP8 guidelines

### PR DESCRIPTION
**Pull request by ShikharJ**
As always please review and suggest improvements.

### CODE REVIEW

**Comment by ShikharJ** [comment]
Ping @isuruf.

File: symengine/compatibility.py [hunk 1]
```diff
@@ -7,6 +7,11 @@
 
 import operator
 from collections import defaultdict
+import sys
+import os
+from collections import namedtuple
+from functools import update_wrapper
+from threading import RLock
```
Comment (isuruf) [inline symengine/compatibility.py]: It's better to keep this file in sync with `sympy/core/compatibility.py`

File: symengine/tests/test_lambdify.py [hunk 2]
```diff
@@ -46,38 +46,40 @@ def allclose(vec1, vec2, rtol=1e-13, atol=1e-13):
 def test_Lambdify():
     n = 7
     args = x, y, z = se.symbols('x y z')
-    l = se.Lambdify(args, [x+y+z, x**2, (x-y)/z, x*y*z], backend='lambda')
-    assert allclose(l(range(n, n+len(args))),
+    L = se.Lambdify(args, [x+y+z, x**2, (x-y)/z, x*y*z], backend='lambda')
```
Comment (isuruf) [inline symengine/tests/test_lambdify.py]: Do you know why this change is needed?

File: symengine/tests/test_series_expansion.py [hunk 3]
```diff
@@ -1,22 +1,23 @@
 from symengine.utilities import raises
 from symengine.lib.symengine_wrapper import (series, have_piranha, have_flint,
-    Symbol, Integer, sin, cos, exp, sqrt, E)
+                                             Symbol, Integer, sin, cos, exp,
+                                             sqrt, E)
```
Comment (isuruf) [inline symengine/tests/test_series_expansion.py]: Is this enforced in PEP 8 ? I like the previous one better, as it has less line.

File: symengine/tests/test_lambdify.py (hunk 2 above)
Comment (ShikharJ) [inline symengine/tests/test_lambdify.py]: It is mentioned [here](https://www.python.org/dev/peps/pep-0008/#names-to-avoid).

File: symengine/tests/test_series_expansion.py (hunk 3 above)
Comment (ShikharJ) [inline symengine/tests/test_series_expansion.py]: The previous format raises issues such relating to under-indented lines.

**Comment by ShikharJ** [comment]
Updated @isuruf. Please take a look.

File: symengine/tests/test_series_expansion.py (hunk 3 above)
Comment (isuruf) [inline symengine/tests/test_series_expansion.py]: PEP 8 is more what you'd call guidelines than actual rules. So you can skip styles that are not appropriate

Comment (ShikharJ) [inline symengine/tests/test_series_expansion.py]: So, should I revert changes that are similar to this?

Comment (isuruf) [inline symengine/tests/test_series_expansion.py]: Yes. Btw, there might be directives for the linter you are using to skip these lines.

**Comment by ShikharJ** [comment]
Updated @isuruf. Please take a look here.

File: symengine/compatibility.py (hunk 1 above)
Comment (isuruf) [inline symengine/compatibility.py]: @ShikharJ, ping

Comment (ShikharJ) [inline symengine/compatibility.py]: Should the entire file remain untouched or should the import related changes be?

Comment (isuruf) [inline symengine/compatibility.py]: Entire file.

File: symengine/tests/test_sympy_conv.py [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
-        sinh, cosh, tanh, coth, asinh, acosh, atanh, acoth, Add, Mul, Pow)
-from symengine.lib.symengine_wrapper import (Subs, Derivative, RealMPFR, ComplexMPC,
-        PyNumber)
+    function_symbol, I, E, pi, exp, gamma, have_mpfr, have_mpc,
+    DenseMatrix, sin, cos, tan, cot, csc, sec, asin, acos, atan,
+    acot, acsc, asec, sinh, cosh, tanh, coth, asinh, acosh, atanh,
+    acoth, Add, Mul, Pow)
+from symengine.lib.symengine_wrapper import (Subs, Derivative,
+                                             RealMPFR, ComplexMPC,
+                                             PyNumber)
```
Comment (isuruf) [inline symengine/tests/test_sympy_conv.py]: Can you fix this as well?

**Comment by ShikharJ** [comment]
Ping @isuruf.
